main/src/mockit/internal/startup/PathToAgentJar.java

```diff
@@ -101,4 +107,151 @@ private String findJarFileContainingThisClass()
+            if (jarEntryResources.hasMoreElements()) {
+search:        do {
+                  // go through the contents and search for "mockit"
+                  BufferedReader jarEntryReader = new BufferedReader(new InputStreamReader(jarEntryUrlCandidate.openStream()));
+                  for (String jarEntryLine = jarEntryReader.readLine(); jarEntryLine != null; jarEntryLine = jarEntryReader.readLine()) {
+                     if (jarEntryLine.toLowerCase().contains("mockit")) {
+                        jarEntryUrl = jarEntryUrlCandidate;
+                        break search;
+                     }
+                  }
+                  // try the next resource
+                  jarEntryUrlCandidate = jarEntryResources.nextElement();
+               } while (jarEntryResources.hasMoreElements());
+            } else {
+               jarEntryUrl = jarEntryUrlCandidate;
+            }
+
+            // no matching resource found
+            if (jarEntryUrl == null) {
+               throw new RuntimeException("No matching resource for JMockit JAR resource '" + jarEntryPath + "' found on classpath");
+            }
+
+            InputStream jarEntryStream = jarEntryUrl.openStream();
+            try
+            {
+               ZipEntry jarEntry = new ZipEntry(jarEntryPath);
+               // set the time to a fixed value, so that the generated JAR does not change each time
+               jarEntry.setTime(0);
+               jos.putNextEntry(jarEntry);
+               for (int count = jarEntryStream.read(buffer); count != -1; count = jarEntryStream.read(buffer))
+               {
+                  jos.write(buffer, 0, count);
+               }
+               jos.closeEntry();
+            }
+            finally
+            {
+               try { jarEntryStream.close(); } catch (IOException ignore) {}
+            }
+         }
+         jos.close();
+
+         // check whether the file is already present, then do not use the newly created file
+         // this prevents overflowing of the temp directory with generated jar files and the try
+         // to replace a jar that may be locked by filesystem means like on Windows
+         byte[] digest = jarDigest.digest();
+         StringBuilder digestStringBuilder = new StringBuilder();
+         for (byte digestByte : digest) {
+            digestStringBuilder.append(format("%02x", digestByte));
+         }
+         File jarFile = new File(System.getProperty("java.io.tmpdir"), format("jmockit-%s.jar", digestStringBuilder));
+         if (!jarFile.exists())
+         {
+            tempJarFile.renameTo(jarFile);
+         }
+
+         return jarFile;
+      }
+      catch (NoSuchAlgorithmException e)
+      {
+         // MD5 is not supported, this is against the specification
+         throw new AssertionError(e);
+      }
+      catch (IOException e)
+      {
+         throw new RuntimeException(e);
+      }
+      finally
+      {
+         if (tempJarFile != null)
```

Comment:
Because the POM says source 1.6 and target 1.6, so no try-with-resources is available unfortunately.